<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.9.1"/>
<title>My Project: feat_extract.FastaFilter Namespace Reference</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<script type="text/javascript">
  $(document).ready(function() { init_search(); });
</script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td style="padding-left: 0.5em;">
   <div id="projectname">My Project
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.9.1 -->
<script type="text/javascript">
var searchBox = new SearchBox("searchBox", "search",false,'Search');
</script>
  <div id="navrow1" class="tabs">
    <ul class="tablist">
      <li><a href="index.html"><span>Main&#160;Page</span></a></li>
      <li class="current"><a href="namespaces.html"><span>Namespaces</span></a></li>
      <li><a href="annotated.html"><span>Classes</span></a></li>
      <li>
        <div id="MSearchBox" class="MSearchBoxInactive">
        <span class="left">
          <img id="MSearchSelect" src="search/mag_sel.png"
               onmouseover="return searchBox.OnSearchSelectShow()"
               onmouseout="return searchBox.OnSearchSelectHide()"
               alt=""/>
          <input type="text" id="MSearchField" value="Search" accesskey="S"
               onfocus="searchBox.OnSearchFieldFocus(true)" 
               onblur="searchBox.OnSearchFieldFocus(false)" 
               onkeyup="searchBox.OnSearchFieldChange(event)"/>
          </span><span class="right">
            <a id="MSearchClose" href="javascript:searchBox.CloseResultsWindow()"><img id="MSearchCloseImg" border="0" src="search/close.png" alt=""/></a>
          </span>
        </div>
      </li>
    </ul>
  </div>
  <div id="navrow2" class="tabs2">
    <ul class="tablist">
      <li><a href="namespaces.html"><span>Namespace&#160;List</span></a></li>
      <li><a href="namespacemembers.html"><span>Namespace&#160;Members</span></a></li>
    </ul>
  </div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

<div id="nav-path" class="navpath">
  <ul>
<li class="navelem"><b>feat_extract</b></li><li class="navelem"><a class="el" href="namespacefeat__extract_1_1_fasta_filter.html">FastaFilter</a></li>  </ul>
</div>
</div><!-- top -->
<div class="header">
  <div class="summary">
<a href="#func-members">Functions</a> &#124;
<a href="#var-members">Variables</a>  </div>
  <div class="headertitle">
<div class="title">feat_extract.FastaFilter Namespace Reference</div>  </div>
</div><!--header-->
<div class="contents">
<table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="func-members"></a>
Functions</h2></td></tr>
<tr class="memitem:ac5365277e337fab6052910b07a8f6087"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="namespacefeat__extract_1_1_fasta_filter.html#ac5365277e337fab6052910b07a8f6087">remove_unknown_AA</a> (s)</td></tr>
<tr class="separator:ac5365277e337fab6052910b07a8f6087"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a6b6a8eca64864960e923d555fb6d81b3"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="namespacefeat__extract_1_1_fasta_filter.html#a6b6a8eca64864960e923d555fb6d81b3">contain_illegals</a></td></tr>
<tr class="separator:a6b6a8eca64864960e923d555fb6d81b3"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:aff7ada2a6f901e17fac38246c970aeff"><td class="memItemLeft" align="right" valign="top">def&#160;</td><td class="memItemRight" valign="bottom"><a class="el" href="namespacefeat__extract_1_1_fasta_filter.html#aff7ada2a6f901e17fac38246c970aeff">Get_Dirr_All_Fasta</a></td></tr>
<tr class="separator:aff7ada2a6f901e17fac38246c970aeff"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a7f6b66b5611f58928a75cf2933962044"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a7f6b66b5611f58928a75cf2933962044"></a>
def&#160;</td><td class="memItemRight" valign="bottom"><b>FilterFastaSeqGroups</b></td></tr>
<tr class="separator:a7f6b66b5611f58928a75cf2933962044"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table><table class="memberdecls">
<tr class="heading"><td colspan="2"><h2 class="groupheader"><a name="var-members"></a>
Variables</h2></td></tr>
<tr class="memitem:a26fdc2740bd8fc1865b9550c90a03143"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a26fdc2740bd8fc1865b9550c90a03143"></a>
list&#160;</td><td class="memItemRight" valign="bottom"><b>ILLEGALS</b> = ['B', 'J', 'Z', 'X', 'U', 'O', 'Z']</td></tr>
<tr class="separator:a26fdc2740bd8fc1865b9550c90a03143"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a3d54a1e88ab0d21e0dc444c531b9e7d9"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a3d54a1e88ab0d21e0dc444c531b9e7d9"></a>
tuple&#160;</td><td class="memItemRight" valign="bottom"><b>Dir</b> = str(argv[1])</td></tr>
<tr class="separator:a3d54a1e88ab0d21e0dc444c531b9e7d9"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a36c511ac2efee6ee40eb5446c2283a8f"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a36c511ac2efee6ee40eb5446c2283a8f"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><b>minCounts</b> = 60</td></tr>
<tr class="separator:a36c511ac2efee6ee40eb5446c2283a8f"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:af74ae78b312dd66844670e0461b7beb0"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="af74ae78b312dd66844670e0461b7beb0"></a>
int&#160;</td><td class="memItemRight" valign="bottom"><b>minLength</b> = 40</td></tr>
<tr class="separator:af74ae78b312dd66844670e0461b7beb0"><td class="memSeparator" colspan="2">&#160;</td></tr>
<tr class="memitem:a5ec63cd80b3b31deba4e4b22f12c9432"><td class="memItemLeft" align="right" valign="top"><a class="anchor" id="a5ec63cd80b3b31deba4e4b22f12c9432"></a>
tuple&#160;</td><td class="memItemRight" valign="bottom"><b>FastaFiles</b> = <a class="el" href="namespacefeat__extract_1_1_fasta_filter.html#aff7ada2a6f901e17fac38246c970aeff">Get_Dirr_All_Fasta</a>(Dirr=Dir)</td></tr>
<tr class="separator:a5ec63cd80b3b31deba4e4b22f12c9432"><td class="memSeparator" colspan="2">&#160;</td></tr>
</table>
<a name="details" id="details"></a><h2 class="groupheader">Detailed Description</h2>
<div class="textblock"><pre class="fragment">Go through a given fasta file (later - mod for sets of fastas),
and output a new fasta file, with sequences containg unknown, or non standard AA
removed; too short sequences removed.

Later, can be used to filter sequences whose ID is a classname; (keeping those
 with a minimum amount of examples, e.g. 30+ samples per CATH classs/domain/..)

Look for:
    5.1.2 Iterating over the records in a sequence ﬁle - BioPy Cookbook
    16.1.1 Filtering a sequence ﬁle  (BioPython cookbook)
    Filtering biopython seqio.
    5.5 Writing Sequence Files   - BioPy cookbook

REGEX to remove last letter:
([a-z]+[.][0-9]{1,3}[.][0-9]{1,3})[.][0-9]{1,3}
$1
</pre> </div><h2 class="groupheader">Function Documentation</h2>
<a class="anchor" id="a6b6a8eca64864960e923d555fb6d81b3"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def feat_extract.FastaFilter.contain_illegals </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>seq</em>, </td>
        </tr>
        <tr>
          <td class="paramkey"></td>
          <td></td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>illegals</em> = <code>ILLEGALS</code>&#160;</td>
        </tr>
        <tr>
          <td></td>
          <td>)</td>
          <td></td><td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Used tofilter  U,B,Z,X.. (non standard AA) from a given sequence.
 Returns True if illegals in sequence.
&gt;&gt;&gt;print(contain_illegals('XSC'))
True
</pre> 
</div>
</div>
<a class="anchor" id="aff7ada2a6f901e17fac38246c970aeff"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def feat_extract.FastaFilter.Get_Dirr_All_Fasta </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>Dirr</em> = <code>'.'</code></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Get all FASTA (*.fasta) files from current working directory,
returns a list of files.
If not additional param given, default is in current dir
CURRENTLY - Does not get files from subdirectories.
</pre><pre class="fragment">old -   for f in os.listdir(sys.argv[1]) :
 We could also do:
for file in glob("*.fasta"):
</pre> 
</div>
</div>
<a class="anchor" id="ac5365277e337fab6052910b07a8f6087"></a>
<div class="memitem">
<div class="memproto">
      <table class="memname">
        <tr>
          <td class="memname">def feat_extract.FastaFilter.remove_unknown_AA </td>
          <td>(</td>
          <td class="paramtype">&#160;</td>
          <td class="paramname"><em>s</em></td><td>)</td>
          <td></td>
        </tr>
      </table>
</div><div class="memdoc">
<pre class="fragment">Checks for and Removes unknown-Illegal AA in string/protein sequence.
Does NOT deal with any ambigous or nonstandard AA in the sequence.
Code for 'unknown amino acid' = Z
UNKNOWN_AA = "Z"
What best to do? (Currently whole protein ignored).
Possible: replace modified with standard, and count presence of
nonstandard AA as a feature (add to dict). Remove or replace others?
</pre> 
</div>
</div>
</div><!-- contents -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated on Sun Mar 1 2015 14:56:19 for My Project by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.9.1
</small></address>
</body>
</html>
