\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat}{}\section{feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat Class Reference}
\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat}\index{feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat@{feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat}}
\subsection*{Public Member Functions}
\begin{DoxyCompactItemize}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a3735b241e73d91edf45a4f1edd722ea7}{\+\_\+\+\_\+init\+\_\+\+\_\+}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a2ccaa9437f70532d45069e4eb94c3c21}{Get\+A\+A\+\_\+\+Freq} (self)
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a603ae748d2527d00821a9a1f63f007d9}{count\+To\+Freq} (self, dict\+\_\+counts)
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a7a9fe63c2f412f7644cc0a814cc5be37}{ordered\+\_\+seq\+\_\+pairs}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a437f8c4ea839093b5e33f4c13e894a09}{alphabet\+\_\+prefix} (self, res)
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a158fb3fdf8ae66e27116ca58e25752bf}{Mer\+Count} (self, kmer\+\_\+counts, k)
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a0f80df6810f710824dbaab6e30ae3775}{Getkgram\+Counts}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a8ce3fc3e99eea09835b08d8af39e67c5}{Getkgram\+Freq}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a8b59569502aef30304207fe0303b28c7}{Get\+K\+Mirrors\+Freq}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a76dc57875b17b82b065e4958eb6ec8ed}{K\+Mirrors} (self, k\+\_\+dict)
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a5e9846e49d823597d17b7ce15d959a09}{Get\+C\+T\+D}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ad804cf2dea87e471c03a409a0391b819}{Get\+\_\+\+Simple\+Prot\+Param} (self)
\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a1f46aa7f046055c9ab534234e4ad244e}{}def {\bfseries Get\+Length} (self)\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a1f46aa7f046055c9ab534234e4ad244e}

\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a405aa53e6c8e9b6616739c9e176b8214}{Get\+Flex} (self)
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a48e1171ffa3fa544caf662be9d09b292}{Get\+\_\+\+P\+I} (self)
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ae15ef6f96924ec48e50aca6ea91dcbea}{calculate\+Protein\+Charge}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a9c66b67d86f78d7fd2e9c3f058d07924}{get\+\_\+net\+Charge} (self, P\+H\+\_\+ranges=\mbox{[}3.\+5)
\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a5536d99e2ad4ca79f6fd94fc5a74db5e}{}def {\bfseries Get\+Aliphaticness} (self)\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a5536d99e2ad4ca79f6fd94fc5a74db5e}

\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ac7bf4d24db5ee443363b023dd9a8f120}{Get\+P\+T\+M\+Motifs} (self)
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a3870adebee4ffffdab545c2f79e09dea}{Get\+Cleavage\+Counts}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ae11f6df3b4efbe4e0975089b5a255c7a}{Dict\+\_\+\+Keys\+\_\+prefix} (self, multilevel\+Dict, Prefix\+Str)
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a58835d011020b381bdd83d77220d38e9}{Get\+\_\+\+Param\+Scales\+\_\+\+Features}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a41c12ab62255724b9aa96a04700655df}{Get\+\_\+\+Sub\+Seq\+Param\+Scales\+\_\+\+Features}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_af7aeb76a124e0b2d6437c62525d76917}{Get\+\_\+\+Param\+Scales}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a55fee91bd4a79b007cbacc83d20b70f7}{Binary\+Autocorrellation}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a24e9a7e4395849f24708b5c29cd61eb0}{Get\+Entropy}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ac133108c6d07aa0f3037758307df04a2}{Entropy\+\_\+\+Seq} (self)
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a06c37ba385c6fa7e877e27a986c273ee}{tail\+\_\+properties}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_aaffaab7c26455e14f5317d574a428e1c}{Get\+Simple\+Features}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a15b15dca5d5f7374c06f6d27fd11699a}{cysteine\+Motif}
\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a4bea8b23af186ae22772ad19d1155206}{}def {\bfseries cysteine\+Space\+Motif} (self)\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a4bea8b23af186ae22772ad19d1155206}

\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a63273ac1c666696e0c4a3d5abb4480eb}{get\+Cysteine\+Motifs}
\item 
def \hyperlink{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ade3d514c79ce995aac26b9744212a9f5}{get\+F\+I\+Disorder}
\end{DoxyCompactItemize}
\subsection*{Public Attributes}
\begin{DoxyCompactItemize}
\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ab51cd86b2a143cc282cb98c683c71ce7}{}{\bfseries seq}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ab51cd86b2a143cc282cb98c683c71ce7}

\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a27db3d8dd292aa26eb55f956e86e1811}{}{\bfseries alph\+\_\+set}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a27db3d8dd292aa26eb55f956e86e1811}

\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a6df19c0a636162d28b14caa40525c658}{}{\bfseries Is\+Standard\+Alph}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a6df19c0a636162d28b14caa40525c658}

\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a8b9dc56ea73c50510370d8959c26877e}{}{\bfseries alph}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a8b9dc56ea73c50510370d8959c26877e}

\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a0ac264af452c71a9ccc1378ea1d6ebbb}{}{\bfseries H\+A\+S\+\_\+\+N}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a0ac264af452c71a9ccc1378ea1d6ebbb}

\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_afd5442aeaafb0e6f716c8ab50ddfd07e}{}{\bfseries H\+A\+S\+\_\+\+C}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_afd5442aeaafb0e6f716c8ab50ddfd07e}

\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a770619a25f9d98d731d9663530d8af4a}{}{\bfseries length}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a770619a25f9d98d731d9663530d8af4a}

\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a6cf971f44583e5997197aa8f9fc19616}{}{\bfseries A\+A\+\_\+\+Counts}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a6cf971f44583e5997197aa8f9fc19616}

\item 
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a5322bfd7836ce8195b6d2c9e73468d05}{}{\bfseries Bio\+\_\+\+Prot\+Param}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a5322bfd7836ce8195b6d2c9e73468d05}

\end{DoxyCompactItemize}


\subsection{Detailed Description}
\begin{DoxyVerb}The GetProDes/ProtFeat class collects most sequence features  calculation modules in a class.
Different features  are returned in dictionary-(like) forms.\end{DoxyVerb}
 

\subsection{Constructor \& Destructor Documentation}
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a3735b241e73d91edf45a4f1edd722ea7}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!\+\_\+\+\_\+init\+\_\+\+\_\+@{\+\_\+\+\_\+init\+\_\+\+\_\+}}
\index{\+\_\+\+\_\+init\+\_\+\+\_\+@{\+\_\+\+\_\+init\+\_\+\+\_\+}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{\+\_\+\+\_\+init\+\_\+\+\_\+}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+\_\+\+\_\+init\+\_\+\+\_\+ (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{Protein\+Sequence = {\ttfamily \textquotesingle{}\textquotesingle{}}, }
\item[{}]{alph\+\_\+letters = {\ttfamily \textquotesingle{}AA20\textquotesingle{}}, }
\item[{}]{H\+A\+S\+\_\+\+N = {\ttfamily True}, }
\item[{}]{H\+A\+S\+\_\+\+C = {\ttfamily True}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a3735b241e73d91edf45a4f1edd722ea7}
\begin{DoxyVerb}Input a protein sequence.

ProteinSequence = string, protein's sequence.
(alph) - alph_letters = string of all the amino acids in the alphabet.
 Default is 20 (normal).Required when getting N-grams/K-mer counts,
 so as to count consistently.

 Feature keys/Values must be consistent in training and test! (Or MADE so post-hoc)

 HAS_N = "Does sequence have N_terminal end" (Or is it a 'subsequence').
 HAS_N = "Does sequence have C_terminal end".

 We may want to have called features update the class instance,
 when the feature is "unchanging" ,i.e  AAC, PI.., in addition to
 returning the feature and its values as a dict.\end{DoxyVerb}
 

\subsection{Member Function Documentation}
\hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a437f8c4ea839093b5e33f4c13e894a09}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!alphabet\+\_\+prefix@{alphabet\+\_\+prefix}}
\index{alphabet\+\_\+prefix@{alphabet\+\_\+prefix}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{alphabet\+\_\+prefix}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+alphabet\+\_\+prefix (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{res}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a437f8c4ea839093b5e33f4c13e894a09}
\begin{DoxyVerb}if non standard/reduced alphabet is used,
then modify feature names (keys in given dictionary)
 of results to have appropiate prefix.

 res = dict of features. Keys = feature names (strings), values: features val.
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a55fee91bd4a79b007cbacc83d20b70f7}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Binary\+Autocorrellation@{Binary\+Autocorrellation}}
\index{Binary\+Autocorrellation@{Binary\+Autocorrellation}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Binary\+Autocorrellation}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Binary\+Autocorrellation (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{signal\+\_\+letters = {\ttfamily \mbox{[}\textquotesingle{}K\textquotesingle{}}, }
\item[{}]{R, }
\item[{}]{loc = {\ttfamily -\/1}, }
\item[{}]{Length\+Normalization = {\ttfamily None}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a55fee91bd4a79b007cbacc83d20b70f7}
\begin{DoxyVerb}Calculate autocorrelation for sequence.
signal_letters - Convert these letter(s) to "1" , and rest
of letters in sequence to 0, then calc autocor as in signal analysis.

http://stackoverflow.com/questions/643699/how-can-i-use-numpy-correlate-to-do-autocorrelation
http://stackoverflow.com/questions/13439718/how-to-interpret-numpy-correlate-and-numpy-corrcoef-values?rq=1

#How to Normalize - By Sum ("1's") or seq.length?
lengthNorm = "length" normalization variable.
     By seq.length ("length") or sum of "1"s/signals ("sum").

Loc = Normalization factor,  By Sum ("1's") or length.?
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ae15ef6f96924ec48e50aca6ea91dcbea}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!calculate\+Protein\+Charge@{calculate\+Protein\+Charge}}
\index{calculate\+Protein\+Charge@{calculate\+Protein\+Charge}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{calculate\+Protein\+Charge}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+calculate\+Protein\+Charge (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{p\+H = {\ttfamily 7.2}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ae15ef6f96924ec48e50aca6ea91dcbea}
\begin{DoxyVerb}Get Net-Protein charge at given PH. Can alse be Used to predict PI of a protein.
Be careful if used on a "partial" seq - +-N/C termini..
If using only part of a sequence (Just the N-tail etc'), then modify call appropaitely.
http://www.petercollingridge.co.uk/sites/files/peter/predictPI.txt
We could also try using:
http://pythonhosted.org/pyteomics/_modules/pyteomics/electrochem.html#charge
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a603ae748d2527d00821a9a1f63f007d9}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!count\+To\+Freq@{count\+To\+Freq}}
\index{count\+To\+Freq@{count\+To\+Freq}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{count\+To\+Freq}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+count\+To\+Freq (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{dict\+\_\+counts}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a603ae748d2527d00821a9a1f63f007d9}
\begin{DoxyVerb}Gets percentage from a provided dict of AA counts.
Could be optimized!.. (Use a list..)
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a15b15dca5d5f7374c06f6d27fd11699a}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!cysteine\+Motif@{cysteine\+Motif}}
\index{cysteine\+Motif@{cysteine\+Motif}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{cysteine\+Motif}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+cysteine\+Motif (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{seg\+Divide = {\ttfamily 3}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a15b15dca5d5f7374c06f6d27fd11699a}
\begin{DoxyVerb}CHANGED: Threshhold for counts removed, made into variable feature instead of discrete
Note: We could re-increase the threshhold (min 1 count instead of >1).
And also then divide into more segs..
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ae11f6df3b4efbe4e0975089b5a255c7a}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Dict\+\_\+\+Keys\+\_\+prefix@{Dict\+\_\+\+Keys\+\_\+prefix}}
\index{Dict\+\_\+\+Keys\+\_\+prefix@{Dict\+\_\+\+Keys\+\_\+prefix}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Dict\+\_\+\+Keys\+\_\+prefix}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Dict\+\_\+\+Keys\+\_\+prefix (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{multilevel\+Dict, }
\item[{}]{Prefix\+Str}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ae11f6df3b4efbe4e0975089b5a255c7a}
\begin{DoxyVerb}given a dict, Returns a new dict with prefix_str added before each key.
Also used when using "alternative" alphs (so feature names will be unique,
    i.e "AA_14_KR Freq" vs "KR Freq" (for self.alph==AA_20))
i.e:

>>> aa_dict = {'fruit':'orange','K':0.23}
>>>  print(cds.transform(aa_dict,"entropy"))
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ac133108c6d07aa0f3037758307df04a2}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Entropy\+\_\+\+Seq@{Entropy\+\_\+\+Seq}}
\index{Entropy\+\_\+\+Seq@{Entropy\+\_\+\+Seq}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Entropy\+\_\+\+Seq}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Entropy\+\_\+\+Seq (
\begin{DoxyParamCaption}
\item[{}]{self}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ac133108c6d07aa0f3037758307df04a2}
\begin{DoxyVerb}DEPRECATE! Use GetEntropy Instead!!

Calculate the (Shannon) information entropy  of a given input string.
Entropy is the expected value of the measure of information content in system.
http://rosettacode.org/wiki/Entropy#Python

Qu - For later comparing to Max Entropy - use possible AA counts?
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a9c66b67d86f78d7fd2e9c3f058d07924}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!get\+\_\+net\+Charge@{get\+\_\+net\+Charge}}
\index{get\+\_\+net\+Charge@{get\+\_\+net\+Charge}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{get\+\_\+net\+Charge}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+get\+\_\+net\+Charge (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{P\+H\+\_\+ranges = {\ttfamily \mbox{[}3.5}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a9c66b67d86f78d7fd2e9c3f058d07924}
\begin{DoxyVerb}SLOW!
Return net charge of sequence for range of PH.
Some (Wiki) PH ranges - Lysosomes - 4.5, Granules of chromaffin cells - 5.5,
Cytosol 7.2, Blood 7.34–7.45, Mitochondrial matrix - 7.5,
Pancreas secretions 8.1
http://en.wikipedia.org/wiki/PH#Living_systems

PH Of Stress response/oxidization (which can trigger unfolding in some proteins) ?
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_af7aeb76a124e0b2d6437c62525d76917}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+\_\+\+Param\+Scales@{Get\+\_\+\+Param\+Scales}}
\index{Get\+\_\+\+Param\+Scales@{Get\+\_\+\+Param\+Scales}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+\_\+\+Param\+Scales}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+\_\+\+Param\+Scales (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{window = {\ttfamily 7}, }
\item[{}]{edge = {\ttfamily 1.0}, }
\item[{}]{Pick\+Scales = {\ttfamily None}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_af7aeb76a124e0b2d6437c62525d76917}
\begin{DoxyVerb}Gets numerical represention of sequence,
for each amino acid propensity scale (and window size). This is then
used (later) to Get values for sequences using different amino propensities,
via def Get_paramScales.

Default is "built in" scales, but can be expanded easily.

Default window size from literature is !17-19 for detecting TMDs' using hydrophobicity.

Returns a list of (string:list) tuples.
Uses:  Bio.SeqUtils.ProtParam.ProteinAnalysis(protein_scale(self, param_dict, window))
http://biopython.org/DIST/docs/api/Bio.SeqUtils.ProtParamData-module.html - builtin scales

TODO: input "scales to use", "select which".
TODO: Copy AA Scales from Bio.SeqUtils.ProtParamData to AAlphabets.py !
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a58835d011020b381bdd83d77220d38e9}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+\_\+\+Param\+Scales\+\_\+\+Features@{Get\+\_\+\+Param\+Scales\+\_\+\+Features}}
\index{Get\+\_\+\+Param\+Scales\+\_\+\+Features@{Get\+\_\+\+Param\+Scales\+\_\+\+Features}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+\_\+\+Param\+Scales\+\_\+\+Features}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+\_\+\+Param\+Scales\+\_\+\+Features (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{window = {\ttfamily 7}, }
\item[{}]{edge = {\ttfamily 1.0}, }
\item[{}]{Pick\+Scales = {\ttfamily None}, }
\item[{}]{trim = {\ttfamily 5}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a58835d011020b381bdd83d77220d38e9}
\begin{DoxyVerb}Extract features for given scale/propensity represention of a protein sequence.
Calls on def Get_ParamScales.

Default is to get all "preloaded" scales.
User can choose to use only a limited number of scales, using PickScales

Get 3: max, min, average for each scale.

trim : get average of top/bottom "trim"'%' of values, and av. of in between values.
(i.e. _AverageTrimMAX = average of top 20%,AverageTrimMIN - bottom 20%, and trimAv = Average of remaining 60%)
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a48e1171ffa3fa544caf662be9d09b292}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+\_\+\+P\+I@{Get\+\_\+\+P\+I}}
\index{Get\+\_\+\+P\+I@{Get\+\_\+\+P\+I}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+\_\+\+P\+I}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+\_\+\+P\+I (
\begin{DoxyParamCaption}
\item[{}]{self}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a48e1171ffa3fa544caf662be9d09b292}
\begin{DoxyVerb}Get predicted PI. (Uses BioPython.ProtParam util)
NOTE! For a "parsed"/cleaved subsequence,
This is innaccurate unless fixed! (Doesn't take lack of N/C terminal into account)
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ad804cf2dea87e471c03a409a0391b819}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+\_\+\+Simple\+Prot\+Param@{Get\+\_\+\+Simple\+Prot\+Param}}
\index{Get\+\_\+\+Simple\+Prot\+Param@{Get\+\_\+\+Simple\+Prot\+Param}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+\_\+\+Simple\+Prot\+Param}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+\_\+\+Simple\+Prot\+Param (
\begin{DoxyParamCaption}
\item[{}]{self}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ad804cf2dea87e471c03a409a0391b819}
\begin{DoxyVerb}Get basic physical properties as in BioPython/ExPasy ProtParam
module.
Returns: PI, MW, GRAVY, aromaticity, Instability index.

Note: These methods all assume a standard AA Alphabet.
Warning! Returned PI is INNACCURATE For a parsed (Tail(s) removed) subseq.
(BioPy-ProtParam.isoelectric_point assumes N,C terminii!)
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a41c12ab62255724b9aa96a04700655df}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+\_\+\+Sub\+Seq\+Param\+Scales\+\_\+\+Features@{Get\+\_\+\+Sub\+Seq\+Param\+Scales\+\_\+\+Features}}
\index{Get\+\_\+\+Sub\+Seq\+Param\+Scales\+\_\+\+Features@{Get\+\_\+\+Sub\+Seq\+Param\+Scales\+\_\+\+Features}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+\_\+\+Sub\+Seq\+Param\+Scales\+\_\+\+Features}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+\_\+\+Sub\+Seq\+Param\+Scales\+\_\+\+Features (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{window = {\ttfamily 5}, }
\item[{}]{edge = {\ttfamily 1.0}, }
\item[{}]{Pick\+Scales = {\ttfamily MinScales\+\_\+Dict}, }
\item[{}]{segs = {\ttfamily 3}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a41c12ab62255724b9aa96a04700655df}
\begin{DoxyVerb}Similar to "Get_ParamScales_Features", but gets a small(er) set of features (per scale),
for a smaller set (by efault) of AA scales, extracted from multiple segments of the sequence.

Extract features for given scale/propensity represention of a protein sequence.
By default, gets less features (min, max, average), with a minSet of scales. (Includes Atchley).
An alternative, novel implementation is to divide into unequal segments -
first,last 20%, middle 60% (or middle 30% *2). Dan Ofer.
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a2ccaa9437f70532d45069e4eb94c3c21}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+A\+A\+\_\+\+Freq@{Get\+A\+A\+\_\+\+Freq}}
\index{Get\+A\+A\+\_\+\+Freq@{Get\+A\+A\+\_\+\+Freq}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+A\+A\+\_\+\+Freq}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+A\+A\+\_\+\+Freq (
\begin{DoxyParamCaption}
\item[{}]{self}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a2ccaa9437f70532d45069e4eb94c3c21}
\begin{DoxyVerb}amino acid composition (20).
Usage:

result = GetAA_Freq()
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a3870adebee4ffffdab545c2f79e09dea}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+Cleavage\+Counts@{Get\+Cleavage\+Counts}}
\index{Get\+Cleavage\+Counts@{Get\+Cleavage\+Counts}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+Cleavage\+Counts}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+Cleavage\+Counts (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{Alt\+Length\+Normalize = {\ttfamily False}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a3870adebee4ffffdab545c2f79e09dea}
\begin{DoxyVerb}Get various Potential sites (and/or regions/products),
 associated with cleavage of a precursor sequence/pre-pro-protein into
 products.
 EG: Neuropeptide Precursor processing.
 Many different potential cleavage motifs, dep. on organism, enzymes..!
 Default is Known motif model (Neuropeptides Precursor cleavage via flanking dibasic).:
 Southey et al. (2006b)  Known Motif model comprised of several prevalent motifs
 associated with neuropeptide precursor cleavage:
Xxx-Xxx-Lys-Lys#,
Xxx-Xxx-Lys-Arg#,
Xxx-Xxx-Arg-Arg#,
Arg-Xxx-Xxx-Lys#,
Arg-Xxx-Xxx-Arg#.
  (Xxx = Any AA. # - Cleavage start site. Lys = K. Arg = R)
    Note - In many cases, Proline should NOT be present prior or adjacent to the cl)

 TODO: "Extract" all putative cleaved products/regions, and analyze them
 using AA propensity scales, etc'  (hydrophobicity, flexibility, products...).
 TODO: More sophisticated Furin Cleavage model suggested in:
 "A 20 Residues Motif Delineates the Furin Cleavage Site[...]"
http://www.nuolan.net/motif.html

    TODO: Use external predictor to filter false positives.
Eg: ELM. (Disorder,SS..)
    TODO: Option to input list of RegEx motifs. For other PC (Pre-convertases. not just furin..)

NOTE: Currently, doesn't normalize for length..
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a5e9846e49d823597d17b7ce15d959a09}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+C\+T\+D@{Get\+C\+T\+D}}
\index{Get\+C\+T\+D@{Get\+C\+T\+D}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+C\+T\+D}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+C\+T\+D (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{ctd = {\ttfamily \textquotesingle{}CTD\textquotesingle{}}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a5e9846e49d823597d17b7ce15d959a09}
\begin{DoxyVerb}Get Composition, Transition, Distribution descriptors.
(Currently uses CTD_pro - external module).
ctd = string of letters, according to which C,T,D will be called.

Usage:
result = GetCTD() #Returns all C,T,D
composition = GetCTD('c') #Returns all C.
Comp_And_Distribution =  = GetCTD('CD') #Returns all C and D.
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a63273ac1c666696e0c4a3d5abb4480eb}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!get\+Cysteine\+Motifs@{get\+Cysteine\+Motifs}}
\index{get\+Cysteine\+Motifs@{get\+Cysteine\+Motifs}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{get\+Cysteine\+Motifs}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+get\+Cysteine\+Motifs (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{seg\+Divide = {\ttfamily 3}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a63273ac1c666696e0c4a3d5abb4480eb}
\begin{DoxyVerb}Get Cysteine spacer motif, and counts of frequent CxxC motifs over protein sequence
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a24e9a7e4395849f24708b5c29cd61eb0}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+Entropy@{Get\+Entropy}}
\index{Get\+Entropy@{Get\+Entropy}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+Entropy}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+Entropy (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{normalize\+Tot\+Entropy = {\ttfamily False}, }
\item[{}]{get\+Letters\+Entropy = {\ttfamily True}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a24e9a7e4395849f24708b5c29cd61eb0}
\begin{DoxyVerb}http://bugra.github.io/work/notes/2014-05-16/entropy-perplexity-image-text/
VS Entropy_Kap_AA , Entropy_seq..

normalizeTotEntropy : If True, then divide total sequence entropy by log2(length).
                (Check if correct!!)
getLettersEntropy : If True, also return  entropy per letter; If false, then
return only the "total entropy".
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ade3d514c79ce995aac26b9744212a9f5}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!get\+F\+I\+Disorder@{get\+F\+I\+Disorder}}
\index{get\+F\+I\+Disorder@{get\+F\+I\+Disorder}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{get\+F\+I\+Disorder}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+get\+F\+I\+Disorder (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{segments = {\ttfamily 5}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ade3d514c79ce995aac26b9744212a9f5}
\begin{DoxyVerb}Divide protein sequence into segments, and returns for each seg.
predicted disorder (Y/N) according to FoldIndex method (Uversky et al).
Method implemented with help of Tal Arian.
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a405aa53e6c8e9b6616739c9e176b8214}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+Flex@{Get\+Flex}}
\index{Get\+Flex@{Get\+Flex}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+Flex}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+Flex (
\begin{DoxyParamCaption}
\item[{}]{self}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a405aa53e6c8e9b6616739c9e176b8214}
\begin{DoxyVerb}Get parameters for B-values / flexibility. Built for a 9 window window;
alters values from a BioPy ProtParam module.
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a0f80df6810f710824dbaab6e30ae3775}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Getkgram\+Counts@{Getkgram\+Counts}}
\index{Getkgram\+Counts@{Getkgram\+Counts}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Getkgram\+Counts}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Getkgram\+Counts (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{k = {\ttfamily 2}, }
\item[{}]{Use\+Variable\+Combinations = {\ttfamily False}, }
\item[{}]{Ignore\+Alphabet\+Type = {\ttfamily False}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a0f80df6810f710824dbaab6e30ae3775}
\begin{DoxyVerb}k-mer peptide composition counts (alphSize^k)

If UseVariableCombinations=False, then calls on a dictionary
of ALL possible k_mer combinations using def NGram_combinations,
(In order to ensure feature count consistency).
This could be skipped, with CARE; (If feature matrix is later kept 'fixed')

Usage:

result = GetkmerFreq()

NOTE: Ideally - n-gram multiLetterComb output dictionary should be saved/pickled,
rather than requiring that it be recalcualted.
By default, length 2 bigram combinations are available for some alphabets.
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a8ce3fc3e99eea09835b08d8af39e67c5}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Getkgram\+Freq@{Getkgram\+Freq}}
\index{Getkgram\+Freq@{Getkgram\+Freq}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Getkgram\+Freq}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Getkgram\+Freq (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{k, }
\item[{}]{Ignore\+Alphabet\+Type = {\ttfamily False}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a8ce3fc3e99eea09835b08d8af39e67c5}
\begin{DoxyVerb}Get K-mer frequency/composition.

If IgnoreAlphabetType=True, (IS False by default) then don't add
a prefix of the type of reduced aa alph to the feature keys.
(Useful when further manipulating by key/string )
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a8b59569502aef30304207fe0303b28c7}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+K\+Mirrors\+Freq@{Get\+K\+Mirrors\+Freq}}
\index{Get\+K\+Mirrors\+Freq@{Get\+K\+Mirrors\+Freq}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+K\+Mirrors\+Freq}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+K\+Mirrors\+Freq (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{k = {\ttfamily 2}, }
\item[{}]{Add\+Mirrors\+Prefix = {\ttfamily True}, }
\item[{}]{get\+Freq = {\ttfamily True}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a8b59569502aef30304207fe0303b28c7}
\begin{DoxyVerb}Get kmer mirror frequencies: if we assume mirror images of letters to be identical.
Merges "mirror images" and sums their values into the remaining 'key'/image.
(Palindromes are unaffected)
I.E: we treat '01' == '10'. but, "11","00",  not affected. But 'KR' and 'RK' would be.

This gives us a number of combinations acc. to Combination with Replacement, vs "Permutations with Replacement".
https://docs.python.org/3.3/library/itertools.html#itertools.combinations_with_replacement
http://www.calculatorsoup.com/calculators/discretemathematics/combinationsreplacement.php

So: (Pseudocode):   #Function does not currently support passing it kgramFreq directly.
    original_kfreq=self.GetkgramFreq(k=2)
    print(original_kfreq)
    {'01':0.2,'10':0.7,'00':0.0,'11':0.1}
    k_mirrors = GetKgramMirrorFreq(original_kfreq)
    print(k_mirrors)
    {'10':6,'00':0.0,'11':0.1}

For k=4, eg: 0111 would me merged with 1110..

Can be configured to get counts rather than frequencies. (getFreq)\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ac7bf4d24db5ee443363b023dd9a8f120}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+P\+T\+M\+Motifs@{Get\+P\+T\+M\+Motifs}}
\index{Get\+P\+T\+M\+Motifs@{Get\+P\+T\+M\+Motifs}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+P\+T\+M\+Motifs}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+P\+T\+M\+Motifs (
\begin{DoxyParamCaption}
\item[{}]{self}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_ac7bf4d24db5ee443363b023dd9a8f120}
\begin{DoxyVerb}Counts occurences of possible PTM motifs.
This is a very weak feature, should be improved using
local sequence properties and external subsequence learning
(average regional flexibility, hydrophobicity, exposure, etc').

We will likely want to replace or augment this with EXTERNAL Predictors.

List of PTM motifs - ELM. (Can also do prediction, accounts for disorder, accessability etc')
http://www.modpred.org/
other predictors..\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_aaffaab7c26455e14f5317d574a428e1c}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Get\+Simple\+Features@{Get\+Simple\+Features}}
\index{Get\+Simple\+Features@{Get\+Simple\+Features}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Get\+Simple\+Features}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Get\+Simple\+Features (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{Param\+Scale\+Window = {\ttfamily 6}, }
\item[{}]{seg\+Divide = {\ttfamily 1}, }
\item[{}]{Disorder\+Segments = {\ttfamily 4}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_aaffaab7c26455e14f5317d574a428e1c}
\begin{DoxyVerb}Returns A large number of "default" features, good for most cases.
This is meant to be called when the sequence is
using the normal 20 letter alphabet. (Otherwise, physical
    parameters returned will be inaccurrate).\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a76dc57875b17b82b065e4958eb6ec8ed}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!K\+Mirrors@{K\+Mirrors}}
\index{K\+Mirrors@{K\+Mirrors}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{K\+Mirrors}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+K\+Mirrors (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{k\+\_\+dict}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a76dc57875b17b82b065e4958eb6ec8ed}
\begin{DoxyVerb}Given a kmer (k_freq?) dict, returns it with mirror images merged and summed.
(Palindromes unaffected).
 = Implementation part of GetKMirrorsFreq
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a158fb3fdf8ae66e27116ca58e25752bf}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!Mer\+Count@{Mer\+Count}}
\index{Mer\+Count@{Mer\+Count}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{Mer\+Count}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+Mer\+Count (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{kmer\+\_\+counts, }
\item[{}]{k}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a158fb3fdf8ae66e27116ca58e25752bf}
\begin{DoxyVerb}Counts all occurences of given k-mers (From the
    kmers dictionary's keys)
in the sequence.
Stores counts as values.
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a7a9fe63c2f412f7644cc0a814cc5be37}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!ordered\+\_\+seq\+\_\+pairs@{ordered\+\_\+seq\+\_\+pairs}}
\index{ordered\+\_\+seq\+\_\+pairs@{ordered\+\_\+seq\+\_\+pairs}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{ordered\+\_\+seq\+\_\+pairs}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+ordered\+\_\+seq\+\_\+pairs (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{distance = {\ttfamily 1}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a7a9fe63c2f412f7644cc0a814cc5be37}
\begin{DoxyVerb}Returns sequence as pairs of bigrams. (Useful for counting).
This function returns (sequentially ordered) letter pairs from seq that
occur at the given distance from each other.

The distance should be an integer larger then 0.

An empty list will be returned if distance is equal to or larger than the
sequence length.

Args:
    seq (str): The sequence of which the pairs are returned
    distance (int): distance between the letter pairs
Returns:
    [(str, str)] List of tuples, each tuple containing two letters of one
         pair. The lengt of the list is max(0, len(seq) - 1).

>>> ordered_seq_pairs('ABCDE', 1)
['AB', 'BC', 'CD', 'DE']
>>> ordered_seq_pairs('ABCDE', 3)
['AD', 'BE']
>>> ordered_seq_pairs('ABCDE', 5)
[]
\end{DoxyVerb}
 \hypertarget{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a06c37ba385c6fa7e877e27a986c273ee}{}\index{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}!tail\+\_\+properties@{tail\+\_\+properties}}
\index{tail\+\_\+properties@{tail\+\_\+properties}!feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat@{feat\+\_\+extract\+::\+Prot\+Feat\+::\+Prot\+Feat}}
\subsubsection[{tail\+\_\+properties}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Prot\+Feat.\+Prot\+Feat.\+tail\+\_\+properties (
\begin{DoxyParamCaption}
\item[{}]{self, }
\item[{}]{tail\+\_\+end = {\ttfamily \textquotesingle{}N\textquotesingle{}}, }
\item[{}]{Get\+Full\+A\+A\+Freq = {\ttfamily True}, }
\item[{}]{reduced\+\_\+alph = {\ttfamily \textquotesingle{}Ofer\+\_\+N\+\_\+Tail\textquotesingle{}}}
\end{DoxyParamCaption}
)}\label{classfeat__extract_1_1_prot_feat_1_1_prot_feat_a06c37ba385c6fa7e877e27a986c273ee}
\begin{DoxyVerb}Method that gets a number of N or C end terminii tail/subsequence's
properties.
IMPORTANT! This method assumed that the self object/protein constructed
 is JUST the tail!
 (I.E - it is NOT meant to be used on part of an instance's own sequence,
    but to be called in the subsequence as its own, constructd object/).

Note: Differences in calling netCharge, paramscale...
PH is currently innacurate in assuming a terminii end.
Assume N-length as being fixed (~25). Likely reduced in its AA alphabet.
Consider whether to use AA frequency or counts.

Signal Peptides in Euk are usually ~15-25 AA long,
often composed of 3 subsections: "charged+"-"Hydrophobic"-"Polar(uncharged)".
\end{DoxyVerb}
 

The documentation for this class was generated from the following file\+:\begin{DoxyCompactItemize}
\item 
Prot\+Feat.\+py\end{DoxyCompactItemize}
