\hypertarget{namespacefeat__extract_1_1_fasta_filter}{}\section{feat\+\_\+extract.\+Fasta\+Filter Namespace Reference}
\label{namespacefeat__extract_1_1_fasta_filter}\index{feat\+\_\+extract.\+Fasta\+Filter@{feat\+\_\+extract.\+Fasta\+Filter}}
\subsection*{Functions}
\begin{DoxyCompactItemize}
\item 
def \hyperlink{namespacefeat__extract_1_1_fasta_filter_ac5365277e337fab6052910b07a8f6087}{remove\+\_\+unknown\+\_\+\+A\+A} (s)
\item 
def \hyperlink{namespacefeat__extract_1_1_fasta_filter_a6b6a8eca64864960e923d555fb6d81b3}{contain\+\_\+illegals}
\item 
def \hyperlink{namespacefeat__extract_1_1_fasta_filter_aff7ada2a6f901e17fac38246c970aeff}{Get\+\_\+\+Dirr\+\_\+\+All\+\_\+\+Fasta}
\item 
\hypertarget{namespacefeat__extract_1_1_fasta_filter_a7f6b66b5611f58928a75cf2933962044}{}def {\bfseries Filter\+Fasta\+Seq\+Groups}\label{namespacefeat__extract_1_1_fasta_filter_a7f6b66b5611f58928a75cf2933962044}

\end{DoxyCompactItemize}
\subsection*{Variables}
\begin{DoxyCompactItemize}
\item 
\hypertarget{namespacefeat__extract_1_1_fasta_filter_a26fdc2740bd8fc1865b9550c90a03143}{}list {\bfseries I\+L\+L\+E\+G\+A\+L\+S} = \mbox{[}\textquotesingle{}B\textquotesingle{}, \textquotesingle{}J\textquotesingle{}, \textquotesingle{}Z\textquotesingle{}, \textquotesingle{}X\textquotesingle{}, \textquotesingle{}U\textquotesingle{}, \textquotesingle{}O\textquotesingle{}, \textquotesingle{}Z\textquotesingle{}\mbox{]}\label{namespacefeat__extract_1_1_fasta_filter_a26fdc2740bd8fc1865b9550c90a03143}

\item 
\hypertarget{namespacefeat__extract_1_1_fasta_filter_a3d54a1e88ab0d21e0dc444c531b9e7d9}{}tuple {\bfseries Dir} = str(argv\mbox{[}1\mbox{]})\label{namespacefeat__extract_1_1_fasta_filter_a3d54a1e88ab0d21e0dc444c531b9e7d9}

\item 
\hypertarget{namespacefeat__extract_1_1_fasta_filter_a36c511ac2efee6ee40eb5446c2283a8f}{}int {\bfseries min\+Counts} = 60\label{namespacefeat__extract_1_1_fasta_filter_a36c511ac2efee6ee40eb5446c2283a8f}

\item 
\hypertarget{namespacefeat__extract_1_1_fasta_filter_af74ae78b312dd66844670e0461b7beb0}{}int {\bfseries min\+Length} = 40\label{namespacefeat__extract_1_1_fasta_filter_af74ae78b312dd66844670e0461b7beb0}

\item 
\hypertarget{namespacefeat__extract_1_1_fasta_filter_a5ec63cd80b3b31deba4e4b22f12c9432}{}tuple {\bfseries Fasta\+Files} = \hyperlink{namespacefeat__extract_1_1_fasta_filter_aff7ada2a6f901e17fac38246c970aeff}{Get\+\_\+\+Dirr\+\_\+\+All\+\_\+\+Fasta}(Dirr=Dir)\label{namespacefeat__extract_1_1_fasta_filter_a5ec63cd80b3b31deba4e4b22f12c9432}

\end{DoxyCompactItemize}


\subsection{Detailed Description}
\begin{DoxyVerb}Go through a given fasta file (later - mod for sets of fastas),
and output a new fasta file, with sequences containg unknown, or non standard AA
removed; too short sequences removed.

Later, can be used to filter sequences whose ID is a classname; (keeping those
 with a minimum amount of examples, e.g. 30+ samples per CATH classs/domain/..)

Look for:
    5.1.2 Iterating over the records in a sequence ﬁle - BioPy Cookbook
    16.1.1 Filtering a sequence ﬁle  (BioPython cookbook)
    Filtering biopython seqio.
    5.5 Writing Sequence Files   - BioPy cookbook

REGEX to remove last letter:
([a-z]+[.][0-9]{1,3}[.][0-9]{1,3})[.][0-9]{1,3}
$1
\end{DoxyVerb}
 

\subsection{Function Documentation}
\hypertarget{namespacefeat__extract_1_1_fasta_filter_a6b6a8eca64864960e923d555fb6d81b3}{}\index{feat\+\_\+extract\+::\+Fasta\+Filter@{feat\+\_\+extract\+::\+Fasta\+Filter}!contain\+\_\+illegals@{contain\+\_\+illegals}}
\index{contain\+\_\+illegals@{contain\+\_\+illegals}!feat\+\_\+extract\+::\+Fasta\+Filter@{feat\+\_\+extract\+::\+Fasta\+Filter}}
\subsubsection[{contain\+\_\+illegals}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Fasta\+Filter.\+contain\+\_\+illegals (
\begin{DoxyParamCaption}
\item[{}]{seq, }
\item[{}]{illegals = {\ttfamily ILLEGALS}}
\end{DoxyParamCaption}
)}\label{namespacefeat__extract_1_1_fasta_filter_a6b6a8eca64864960e923d555fb6d81b3}
\begin{DoxyVerb}Used tofilter  U,B,Z,X.. (non standard AA) from a given sequence.
 Returns True if illegals in sequence.
>>>print(contain_illegals('XSC'))
True
\end{DoxyVerb}
 \hypertarget{namespacefeat__extract_1_1_fasta_filter_aff7ada2a6f901e17fac38246c970aeff}{}\index{feat\+\_\+extract\+::\+Fasta\+Filter@{feat\+\_\+extract\+::\+Fasta\+Filter}!Get\+\_\+\+Dirr\+\_\+\+All\+\_\+\+Fasta@{Get\+\_\+\+Dirr\+\_\+\+All\+\_\+\+Fasta}}
\index{Get\+\_\+\+Dirr\+\_\+\+All\+\_\+\+Fasta@{Get\+\_\+\+Dirr\+\_\+\+All\+\_\+\+Fasta}!feat\+\_\+extract\+::\+Fasta\+Filter@{feat\+\_\+extract\+::\+Fasta\+Filter}}
\subsubsection[{Get\+\_\+\+Dirr\+\_\+\+All\+\_\+\+Fasta}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Fasta\+Filter.\+Get\+\_\+\+Dirr\+\_\+\+All\+\_\+\+Fasta (
\begin{DoxyParamCaption}
\item[{}]{Dirr = {\ttfamily \textquotesingle{}.\textquotesingle{}}}
\end{DoxyParamCaption}
)}\label{namespacefeat__extract_1_1_fasta_filter_aff7ada2a6f901e17fac38246c970aeff}
\begin{DoxyVerb}Get all FASTA (*.fasta) files from current working directory,
returns a list of files.
If not additional param given, default is in current dir
CURRENTLY - Does not get files from subdirectories.
\end{DoxyVerb}
\begin{DoxyVerb}old -   for f in os.listdir(sys.argv[1]) :
 We could also do:
for file in glob("*.fasta"):
\end{DoxyVerb}
 \hypertarget{namespacefeat__extract_1_1_fasta_filter_ac5365277e337fab6052910b07a8f6087}{}\index{feat\+\_\+extract\+::\+Fasta\+Filter@{feat\+\_\+extract\+::\+Fasta\+Filter}!remove\+\_\+unknown\+\_\+\+A\+A@{remove\+\_\+unknown\+\_\+\+A\+A}}
\index{remove\+\_\+unknown\+\_\+\+A\+A@{remove\+\_\+unknown\+\_\+\+A\+A}!feat\+\_\+extract\+::\+Fasta\+Filter@{feat\+\_\+extract\+::\+Fasta\+Filter}}
\subsubsection[{remove\+\_\+unknown\+\_\+\+A\+A}]{\setlength{\rightskip}{0pt plus 5cm}def feat\+\_\+extract.\+Fasta\+Filter.\+remove\+\_\+unknown\+\_\+\+A\+A (
\begin{DoxyParamCaption}
\item[{}]{s}
\end{DoxyParamCaption}
)}\label{namespacefeat__extract_1_1_fasta_filter_ac5365277e337fab6052910b07a8f6087}
\begin{DoxyVerb}Checks for and Removes unknown-Illegal AA in string/protein sequence.
Does NOT deal with any ambigous or nonstandard AA in the sequence.
Code for 'unknown amino acid' = Z
UNKNOWN_AA = "Z"
What best to do? (Currently whole protein ignored).
Possible: replace modified with standard, and count presence of
nonstandard AA as a feature (add to dict). Remove or replace others?
\end{DoxyVerb}
 